File: server/Models/InputModels/RegisterModel.cs
using System;
using Microsoft.AspNetCore.Http;
using Server.Models.Interfaces;
using Server.Models.OutputModels;

namespace Server.Models.InputModels
{
  public class RegisterModel : IUSer
  {
    public string UserName { get; set; }
    public string Password { get; set; }
    public DateTime BirthDate { get; set; }
    public string DisplayName { get; set; }
    public string Bio { get; set; }
    public IFormFile ProfilePicture { get; set; }
  }
}

File: client/src/models/auth/register-model.ts
export type RegisterModel = {
  UserName: string;
  Password: string; 
  BirthDate: Date;
  DisplayName: string;
  Bio: string;
  ProfilePicture: File | null;
}

export const createEmptyModel = 
  (): RegisterModel => ({
    UserName: "",
    Password: "",
    BirthDate: new Date(Date.now()),
    DisplayName: "",
    Bio: "",
    ProfilePicture: null,
  });
